cibyl/plugins/openstack/sources/jenkins_job_builder.py

```diff
@@ -147,6 +167,39 @@ def _get_ip_version(self, path, **kwargs):
                 return None
         return ip_version_str
 
+    def _get_release(self, path, **kwargs):
+        """
+        extract release from the JJB xml file and
+        represent it in the form of string, e.g
+           17.0 or 16.2
+
+        Note: this function is used to support filtering
+            e.g. --release 17.0
+        :param path: to JJB xml file
+        :param **kwargs: cibyl command line
+
+        :return: release string
+                 None if filtered out
+        """
+        release_str = ""
+        if "release" in kwargs:
+            in_mem_file = parse_xml(path)
+            result = set([])
+
+            lines = [line.rstrip() for line in in_mem_file if
+                     "rhos" in line or "send_results_to_umb" in line]
+            for line in lines:
+                release_lst = re.findall(RELEASE, line)
+                for el in release_lst:
+                    releases = set(re.findall(RELEASE_NUMBER, el))
+                    result = result.union(releases)
+            release_str = ",".join(result)
+            # filtering support e.g. --release 18
+            if kwargs['release'].value and args_are_in_list(
```

Comment:
Shouldn't the logic here be reversed?, i.e. `if kwargs['release'].value and not args_are_in_list(...)`

Reply:
Fixed